### PR TITLE
Minor SGD docs clarification fixing #137356, #137352

### LINKED ISSUE #137352: `nesterov` parameter of `optim.SGD()` should be explainid more clearly
### 📚 The doc issue

The doc of [optim.SGD()](https://pytorch.org/docs/stable/generated/torch.optim.SGD.html) explains `nesterov` parameter as shown below but it should be explained more clearly:

> nesterov ([bool](https://docs.python.org/3/library/functions.html#bool), optional) – enables Nesterov momentum (default: `False`)

### Suggest a potential alternative/fix

So, it should be explained something like as shown below:

> nesterov ([bool](https://docs.python.org/3/library/functions.html#bool), optional) – If `True`, Nesterov's Momentum is used while if `False`, Momentum is used. (default: `False`)

cc @svekars @brycebortree @sekyondaMeta @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar

### LINKED ISSUE #137356: `dampening` and `weight_decay` parameter should be explained in order
### 📚 The doc issue

[The doc](https://pytorch.org/docs/stable/generated/torch.optim.SGD.html) of `optim.SGD()` shows `dampening` and `weight_decay` parameter in order as shown below:

> Class torch.optim.SGD(..., momentum=0, dampening=0, weight_decay=0, nesterov=False, ...)

But in `Parameters`  section, `dampening` and `weight_decay` parameter are explained in opposite order as shown below:

> Parameters
> - ...
> - momentum ([float](https://docs.python.org/3/library/functions.html#float), optional) – momentum factor (default: 0)
> - weight_decay ([float](https://docs.python.org/3/library/functions.html#float), optional) – weight decay (L2 penalty) (default: 0) <- Here
> - dampening ([float](https://docs.python.org/3/library/functions.html#float), optional) – dampening for momentum (default: 0) <- Here
> - nesterov ([bool](https://docs.python.org/3/library/functions.html#bool), optional) – enables Nesterov momentum (default: False)   
> - ...

### Suggest a potential alternative/fix

So in `Parameters`  section, `dampening` and `weight_decay` parameter should be explained in order as shown below:

> Parameters
> - ...
> - momentum ([float](https://docs.python.org/3/library/functions.html#float), optional) – momentum factor (default: 0)
> - dampening ([float](https://docs.python.org/3/library/functions.html#float), optional) – dampening for momentum (default: 0) <- Here
> - weight_decay ([float](https://docs.python.org/3/library/functions.html#float), optional) – weight decay (L2 penalty) (default: 0) <- Here
> - nesterov ([bool](https://docs.python.org/3/library/functions.html#bool), optional) – enables Nesterov momentum (default: False)   
> - ...

cc @svekars @brycebortree @sekyondaMeta @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar

### PR DESCRIPTION
**Pull request by janeyx99**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137528

### CODE REVIEW

**Review by albanD — APPROVED** [review]
Sure
